### PR TITLE
Retry commands in testament again

### LINKED ISSUE #16255: retry on failure to avoid frequent 443 / 503 / etc github/download errors
either nimble should retry on failure, on nim CI should retry on failure to avoid frequent flaky test failures caused by spurious 443 / 503 / etc github/download errors.

### Example
see some examples here: https://github.com/timotheecour/Nim/issues/266

### Current Output
* Error: unhandled exception: Unable to query remote tags for
* Failure: reInstallFailed
* error: RPC failed

for example in https://github.com/nim-lang/Nim/pull/16242
https://dev.azure.com/nim-lang/Nim/_build/results?buildId=10637&view=logs&jobId=30931762-47c4-53b3-6a83-316eb5a6b9d7&j=ba7bbfa4-f55c-5c34-6d52-1b6b4edd3f37&t=d3579a4b-a806-5abe-ed48-07e8693a62db
```
Downloading https://github.com/Araq/libffi using git
       Tip: 3 messages have been suppressed, use --verbose to show them.
     Error: Execution failed with exit code 128
        ... Command: git clone --recursive --depth 1 -b 1.0.4 https://github.com/Araq/libffi /tmp/nimble_87545/githubcom_Araqlibffi
        ... Output: Cloning into '/tmp/nimble_87545/githubcom_Araqlibffi'...
        ... error: RPC failed; curl 18 transfer closed with outstanding read data remaining
        ... fatal: expected flush after ref listing
```

```
Failure: reInstallFailed
  git clone loloicci/nimly pkgstemp/nimly
  Cloning into 'pkgstemp/nimly'...
  error: RPC failed; curl 18 transfer closed with outstanding read data remaining
  fatal: error reading section header 'shallow-info'
```

### Expected Output
works

### Possible Solution
I had largely eliminated this problem in https://github.com/nim-lang/Nim/pull/14547 with `actionRetry` (with exponential backoff) but `actionRetry` was reverted (without explanation) as part of https://github.com/nim-lang/Nim/pull/15701 (an otherwise good PR). See discussion in https://github.com/nim-lang/Nim/pull/15701#discussion_r513825056

### Additional Information
devel 1.5.1 e4e5a0c65ab10f48746f8f9b2417e8ca69724a04

### links
* https://stackoverflow.com/a/12651825/1426932 (can be caused by multiple errors but restarting should work, as was done in `actionRetry`)

### PR DESCRIPTION
**Pull request by Clyybber**
Addresses https://github.com/nim-lang/Nim/pull/15701#discussion_r513825056
Fixes #16255

### CODE REVIEW

**Review by narimiran — APPROVED** [review]

**Comment by narimiran** [comment]
Note: Add `[backport:1.4]` when merging.

File: testament/categories.nim [hunk 1]
```diff
@@ -482,6 +482,17 @@ proc makeSupTest(test, options: string, cat: Category): TTest =
   result.options = options
   result.startTime = epochTime()
 
+const maxRetries = 3
+template retryCommand(call): untyped =
+  var res: typeof(call)
+  var backoff = 1
+  for i in 0..<maxRetries:
+    res = call
+    if res.exitCode == QuitSuccess or i == maxRetries: break
```
Comment (timotheecour) [inline testament/categories.nim]: @Clyybber this should be `i == maxRetries-1`

Comment (narimiran) [reply testament/categories.nim]:
```suggestion
    if res.exitCode == QuitSuccess or i == maxRetries-1: break
```

**Comment by timotheecour** [comment]
thanks!

**Pull request merged**